# 第10回　ワークブック

財務指標と市場評価：ROE・PBR・PER・デュポン分解

2026年11月10日（火）3限　金融論

## 使い方

上から順に、1つずつ実行していくnotebookです。

- 実行する　code cellを選び、Shift+Enterで実行します。押す前に、何が出るか予想します。
- やってみる　すぐ下の `# ここに書きます` のcellに自分で書いて実行します。既にあるcellの数字を書き換える指示もあります。

errorを確かめるcellは、確かめたら行頭に `#` を付けておきます。3節で全部を実行し直すとき、errorで止まらなくなります。

詰まったら、cellの内容とエラーの表示をCodexに見せて聞けます。「やってみる」の答え合わせも頼めます。

1節は架空の数字だけで計算します。2節では、J-Quants APIからトヨタ自動車の決算と株価を取得します。取得のcodeは第8回・第9回と同じ形で、2-1のcellに書いてあります。実行するたびにAPIから取り直します。取得したデータは、J-Quantsの利用条件により、そのままの形で他の人に配布しません。

## 準備

1. `workbook/` folderを、自分の作業場所にコピーします。
2. VS Codeでfolderごと開きます（File → Open Folder）。
3. このnotebookを開き、右上の Select Kernel から `finance-env` を選びます。
4. 下のcellを実行します。表示されたpathに `finance-env` が含まれていれば準備完了です。

In [1]:
import sys
print(sys.executable)

[検証に使用したPython]


# 1. 架空の会社で計算する

株価1,000円、株数100万株、自己資本5億円の会社です。株数は、発行済株式数から自己株式を引いた数とします。

## 1-1　時価総額、BPS、PBR

実行する

In [2]:
price = 1000            # 株価（円）
shares = 1_000_000      # 株数（株）
equity = 500_000_000    # 自己資本（円）

market_cap = price * shares
bps = equity / shares

print("時価総額:", market_cap)
print("BPS:", bps)
print("PBR（時価総額 / 自己資本）:", market_cap / equity)
print("PBR（株価 / BPS）:", price / bps)

時価総額: 1000000000
BPS: 500.0
PBR（時価総額 / 自己資本）: 2.0
PBR（株価 / BPS）: 2.0


時価総額は10億円、BPS（1株当たり純資産）は500円です。PBRは、会社全体で計算しても1株当たりで計算しても2倍になります。数字の `_` は桁の区切りで、Pythonは無視します。

やってみる

上のcellの `equity` を `400_000_000`（4億円）に変えて実行します。BPSは400円、PBRはどちらの計算でも2.5倍です。確かめたら `500_000_000` に戻し、もう一度実行します。

## 1-2　株式分割

1株を2株に分けます。会社の自己資本は変わりません。

実行する

In [3]:
split = 2
price_after = price / split
shares_after = shares * split
bps_after = equity / shares_after

print("分割後の株価:", price_after)
print("分割後のBPS:", bps_after)
print("分割後の株価 / 分割後のBPS:", price_after / bps_after)
print("分割後の株価 / 分割前のBPS:", price_after / bps)

分割後の株価: 500.0
分割後のBPS: 250.0
分割後の株価 / 分割後のBPS: 2.0
分割後の株価 / 分割前のBPS: 1.0


株価とBPSを同じ株数の基準でそろえると、PBRは2倍のままです。分割後の株価を分割前のBPSで割ると1倍になり、会社の評価が下がったように見えます。

やってみる

下のcellで `split = 3` として同じ計算をします。分割後のBPSは約166.67円、そろえたPBRは2倍、基準を混ぜたPBRは約0.67倍です。

In [4]:
# ここに書きます

## 1-3　ROEと自己資本の大きさ

純利益は5,000万円とします。

実行する

In [5]:
net_income = 50_000_000
print("ROE:", net_income / equity)

equity_small = 250_000_000
print("自己資本が半分のときのROE:", net_income / equity_small)

ROE: 0.1
自己資本が半分のときのROE: 0.2


同じ純利益でも、自己資本が半分ならROEは10%から20%になります。ROEが高い理由は、利益が大きいことだけではありません。

やってみる

純利益が変わらず、自己資本が10億円のときのROEを計算します。5%になります。

In [6]:
# ここに書きます

## 1-4　デュポン分解

売上高10億円、総資産10億円とします。

実行する

In [7]:
sales = 1_000_000_000
total_assets = 1_000_000_000

margin = net_income / sales             # 売上高純利益率
turnover = sales / total_assets         # 総資産回転率
leverage = total_assets / equity        # 財務レバレッジ

print("売上高純利益率:", margin)
print("総資産回転率:", turnover)
print("財務レバレッジ:", leverage)
print("自己資本比率:", equity / total_assets)
print("3つの積:", margin * turnover * leverage)
print("直接計算したROE:", net_income / equity)

売上高純利益率: 0.05
総資産回転率: 1.0
財務レバレッジ: 2.0
自己資本比率: 0.5
3つの積: 0.1
直接計算したROE: 0.1


3つの積と直接計算したROEは、どちらも0.1（10%）です。財務レバレッジは自己資本比率の逆数です。

やってみる

上のcellの `total_assets` を `2_000_000_000`（20億円）に変えて実行します。総資産回転率は0.5回、財務レバレッジは4倍、自己資本比率は0.25になります。ROEは10%のままです。確かめたら `1_000_000_000` に戻して実行します。

## 1-5　期末・期首・平均の自己資本

期首の自己資本を4.7億円とします。4.7億円に純利益0.5億円を足し、配当0.2億円を引くと、期末の5億円になります。

実行する

In [8]:
equity_begin = 470_000_000
equity_avg = (equity_begin + equity) / 2

print("期末の自己資本で割ったROE:", net_income / equity)
print("期首の自己資本で割ったROE:", net_income / equity_begin)
print("平均の自己資本で割ったROE:", net_income / equity_avg)

期末の自己資本で割ったROE: 0.1
期首の自己資本で割ったROE: 0.10638297872340426
平均の自己資本で割ったROE: 0.10309278350515463


同じ会社・同じ年でも、分母の取り方でROEは10%、約10.64%、約10.31%と変わります。どれを使ったかを、表や図に書いておきます。

やってみる

期首の自己資本が4億円だったとして、平均の自己資本で割ったROEを計算します。約11.11%です。

In [9]:
# ここに書きます

## 1-6　EPS・DPSと、PER・PBR・配当利回り

配当の総額は2,000万円とします。

実行する

In [10]:
dividends = 20_000_000

eps = net_income / shares      # 1株当たり純利益
dps = dividends / shares       # 1株当たり配当
per = price / eps
pbr = price / bps
dividend_yield = dps / price

print("EPS:", eps, " DPS:", dps)
print("PER:", per, " PBR:", pbr, " 配当利回り:", dividend_yield)

EPS: 50.0  DPS: 20.0
PER: 20.0  PBR: 2.0  配当利回り: 0.02


EPSは50円、DPSは20円、PERは20倍、PBRは2倍、配当利回りは2%です。

やってみる

上のcellの先頭に `price = 1500` を足して実行します。PERは30倍、PBRは3倍、配当利回りは約1.33%になります。確かめたらその行を消し、1-1のcellから順に実行し直します。

## 1-7　PBR＝PER×ROE

実行する

In [11]:
roe_end = net_income / equity

print("PER × ROE（期末の自己資本）:", per * roe_end)
print("PBR:", pbr)
print("PER × ROE（平均の自己資本）:", per * net_income / equity_avg)

PER × ROE（期末の自己資本）: 2.0
PBR: 2.0
PER × ROE（平均の自己資本）: 2.0618556701030926


期末の自己資本で割ったROEを使うと、PER×ROEはPBRと同じ2倍です。平均の自己資本で割ったROEを使うと約2.06倍になり、一致しません。PBRのBPSが期末の自己資本だからです。

やってみる

1:2の株式分割の後の数字（株価500円、株数200万株）でEPS、BPS、PER、PBRを計算し、PER×ROEがPBRと一致することを確かめます。EPSは25円、BPSは250円、PERは20倍、PBRは2倍です。

In [12]:
# ここに書きます

## 1-8　割引率が違えばPBRも違う

2024年度の株価の教材では、利益 $E$ のうち割合 $Q$ を配当し、投資家が求める収益率を $k$ として、株価を $P = EQ/k$、$\text{PER} = Q/k$ と表しました。利益をすべて配当し（$Q=1$）、利益が毎年同じなら、$\text{PBR} = \text{PER} \times \text{ROE} = \text{ROE}/k$ です。

実行する

In [13]:
def pbr_no_growth(roe, k, payout=1.0):
    """利益が毎年同じで、割合payoutを配当する場合のPBR（PER = payout / k）"""
    return payout / k * roe

print("A社　ROE 10%、k 5%:", round(pbr_no_growth(0.10, 0.05), 2))
print("B社　ROE 15%、k 10%:", round(pbr_no_growth(0.15, 0.10), 2))

A社　ROE 10%、k 5%: 2.0
B社　ROE 15%、k 10%: 1.5


B社のほうがROEは高いのに、投資家が求める収益率も高いので、PBRはA社の2倍より低い1.5倍です。ROEだけでPBRの高さは決まりません。

やってみる

B社の `k` を0.075（7.5%）に下げると、PBRは2倍になります。

In [14]:
# ここに書きます

# 2. トヨタ自動車の決算と株価で計算する

## 2-1　決算サマリーと株価を読み込む

下の1つ目のcellは、第8回・第9回と同じ取得の準備です。`fetch` がURLに条件を付けてJ-Quantsに送り、返ってきたJSONをDataFrameにします。2つ目のcellで、決算サマリーと2026年6月の株価を取得します。実行するたびにAPIへ問い合わせます。最初の実行では、API keyの入力欄が出ます。入力した値は画面にもfileにも残りません。

実行する

In [15]:
import pandas as pd

# 第8回・第9回と同じ仕組みです。URLに条件を付けて送ると、JSONが返ってきます。
# その "data" の中身（1行が1つの辞書）をDataFrameにします。
# API keyは、環境変数 JQUANTS_API_KEY にあればそれを使い、なければ最初の取得のときに
# 非表示の入力欄で受け取ります。このfileにも出力にも書き込みません。
# 実行するたびにAPIから取り直します。開示が追加・訂正されていれば、行や数値が前回と変わります。
# 取得したデータは、J-Quantsの利用条件により、そのままの形で他の人に配布しません。
#
# 応答が200以外のとき fetch は「HTTP 4xx」で止まります。よくある原因は次のとおりです。
#   400  銘柄コード・日付・取得できる期間（Freeプランは12週間前まで）が違う
#   401  API keyが違う
#   403  このプランでは取れないデータか期間
#   429  回数制限。Freeプランは1分に5回まで。少し待ってからやり直す
#
# 行を選んで Shift+Enter で使うときは、先に下の import と BASE・api_key の行を送り、
# 次に def の行から関数の終わりまでをまとめて送ります。関数の中の数行だけを送ると SyntaxError になります。
# 1行が長かったり、複数の関数を一度に選んだりすると、terminalの行編集が途中で崩れることがあります。
# 一度送った定義はそのterminalを閉じるまで残るので、以後は fetch(...) の行だけを送れば済みます。
import os
from getpass import getpass

import requests

BASE = "https://api.jquants.com/v2"
api_key = os.environ.get("JQUANTS_API_KEY", "")


def fetch(endpoint, params):
    """J-Quants API V2に条件を送り、返ってきた全行をDataFrameにする。"""
    global api_key
    if not api_key:
        api_key = getpass("J-Quants API key: ").strip()
    params = dict(params)
    rows = []
    while True:
        r = requests.get(BASE + endpoint, params=params,
                         headers={"x-api-key": api_key}, timeout=30)
        if r.status_code != 200:
            raise RuntimeError(f"HTTP {r.status_code}")
        payload = r.json()
        rows.extend(payload["data"])
        cursor = payload.get("pagination_key")
        if not cursor:
            break
        params["pagination_key"] = cursor
    if not rows:
        raise RuntimeError("0件です。銘柄・期間を確認します。")
    return pd.DataFrame(rows)


In [16]:
code = "72030"   # 証券コード7203の末尾に0を付けた5桁
fins = fetch("/fins/summary", {"code": code})
daily = fetch("/equities/bars/daily",
                      {"code": code, "from": "2026-06-01", "to": "2026-06-19"})

# 読んだ直後はすべて文字列なので、使う列を日付型と数値型に直す（第7回・第9回と同じ手順）
for col in ["DiscDate", "CurPerEn", "CurFYEn"]:
    fins[col] = pd.to_datetime(fins[col], errors="coerce")
for col in ["Sales", "NP", "TA", "Eq", "ShEq", "EqAR", "EPS", "BPS", "DivAnn", "ShOutFY", "TrShFY", "ROE"]:
    fins[col] = pd.to_numeric(fins[col], errors="coerce")
fins = fins.sort_values(["DiscDate", "CurPerEn"]).reset_index(drop=True)
daily["Date"] = pd.to_datetime(daily["Date"])
for col in ["C", "AdjFactor", "AdjC"]:
    daily[col] = pd.to_numeric(daily[col], errors="coerce")

fins[["DiscDate", "DocType", "CurPerType", "CurPerEn", "CurFYEn"]]

,DiscDate,DocType,CurPerType,CurPerEn,CurFYEn
0,2024-08-01,1QFinancialStatements_Consolidated_IFRS,1Q,2024-06-30,2025-03-31
1,2024-11-06,2QFinancialStatements_Consolidated_IFRS,2Q,2024-09-30,2025-03-31
2,2025-02-05,3QFinancialStatements_Consolidated_IFRS,3Q,2024-12-31,2025-03-31
3,2025-05-08,FYFinancialStatements_Consolidated_IFRS,FY,2025-03-31,2025-03-31
4,2025-08-07,1QFinancialStatements_Consolidated_IFRS,1Q,2025-06-30,2026-03-31
5,2025-11-05,2QFinancialStatements_Consolidated_IFRS,2Q,2025-09-30,2026-03-31
6,2026-02-06,3QFinancialStatements_Consolidated_IFRS,3Q,2025-12-31,2026-03-31
7,2026-05-08,FYFinancialStatements_Consolidated_IFRS,FY,2026-03-31,2026-03-31


1行が1回の開示です。四半期（1Q〜3Q）と通期（FY）が混ざっています。金額の列は円単位です。取得した日によって、表の下に新しい開示が増えていることがあります。

## 2-2　通期の行を選ぶ

通期（FY）・連結の決算短信で、評価日2026年6月19日までに開示された行を残します。同じ決算期の開示が2回以上あれば、最後の開示を使います。第11回では、同じ手順を多くの会社に広げます。

実行する

In [17]:
eval_date = "2026-06-19"
fy = fins[(fins["CurPerType"] == "FY")
          & fins["DocType"].str.startswith("FYFinancialStatements_Consolidated_")
          & (fins["DiscDate"] <= eval_date)]
fy = fy.sort_values(["CurPerEn", "DiscDate"]).drop_duplicates("CurPerEn", keep="last")
fy[["DiscDate", "CurFYEn", "DocType", "Sales", "NP", "TA", "Eq", "ShEq", "EPS", "BPS", "DivAnn"]]

,DiscDate,CurFYEn,DocType,Sales,NP,TA,Eq,ShEq,EPS,BPS,DivAnn
3,2025-05-08,2025-03-31,FYFinancialStatements_Consolidated_IFRS,48036704000000,4765086000000,93601350000000,36878913000000,35924826000000,359.56,2753.09,90.0
7,2026-05-08,2026-03-31,FYFinancialStatements_Consolidated_IFRS,50684952000000,3848098000000,105522331000000,41020068000000,39918854000000,295.25,3062.82,95.0


2025年3月期と2026年3月期の2行です。`DocType` の末尾の `IFRS` は、国際会計基準の連結決算であることを示します。`NP` は親会社の所有者に帰属する当期利益、`Eq` は純資産、`ShEq` は自己資本です。

## 2-3　評価日の株価

評価日は2026年6月19日とします。2026年3月期の決算は5月8日に開示されているので、評価日には公表済みです。

実行する

In [18]:
day = daily[daily["Date"] == "2026-06-19"]
price = float(day["C"].iloc[0])
day[["Date", "Code", "C", "AdjFactor", "AdjC"]]

,Date,Code,C,AdjFactor,AdjC
14,2026-06-19,72030,2776.5,1.0,2776.5


終値 `C` は2,776.5円です。この期間には株式分割がないので、`AdjC` も同じ値です。

やってみる

2026年6月1日の終値を表示します。2,905.5円です。

In [19]:
# ここに書きます

## 2-4　自己資本とBPSの定義を確かめる

実行する

In [20]:
if len(fy) < 2: raise ValueError("評価日までの連結通期が2期必要です")
cur, prev = fy.iloc[-1], fy.iloc[-2]
print(cur[["Code","CurPerEn","DiscDate","DocType"]])

shares = cur["ShOutFY"] - cur["TrShFY"]     # 期末発行済株式数 - 期末自己株式数
print("株数:", shares)
print("BPS（決算短信）:", cur["BPS"])
print("ShEq / 株数:", round(cur["ShEq"] / shares, 2))
print("Eq / 株数:", round(cur["Eq"] / shares, 2))

Code                                          72030
CurPerEn                        2026-03-31 00:00:00
DiscDate                        2026-05-08 00:00:00
DocType     FYFinancialStatements_Consolidated_IFRS
Name: 7, dtype: object
株数: 13033384474
BPS（決算短信）: 3062.82
ShEq / 株数: 3062.82
Eq / 株数: 3147.31


決算短信のBPS 3,062.82円は、自己資本 `ShEq` を自己株式を除いた株数で割った値と一致します。純資産 `Eq` で割ると3,147.31円になります。純資産には、子会社の株式のうち親会社以外が持つ部分（非支配持分）が入っているからです。この先は `ShEq` を自己資本として使います。

やってみる

自己資本比率 `EqAR` に総資産 `TA` を掛けて、`ShEq` と比べます。39,887,441,118,000円で、`ShEq` の39,918,854,000,000円より少し小さくなります。`EqAR` が小数第3位で丸めてあるためです。

In [21]:
# ここに書きます

## 2-5　ROEとデュポン分解

実行する

In [22]:
net_income = cur["NP"]
sales = cur["Sales"]
total_assets = cur["TA"]
equity = cur["ShEq"]
equity_prev = prev["ShEq"]

margin = net_income / sales
turnover = sales / total_assets
leverage = total_assets / equity

print("売上高純利益率:", round(margin, 4))
print("総資産回転率:", round(turnover, 4))
print("財務レバレッジ:", round(leverage, 4))
print("3つの積:", round(margin * turnover * leverage, 4))
print("ROE（期末の自己資本）:", round(net_income / equity, 4))
print("ROE（平均の自己資本）:", round(net_income / ((equity_prev + equity) / 2), 4))
print("ROE（決算サマリーの列）:", cur["ROE"])

売上高純利益率: 0.0759
総資産回転率: 0.4803
財務レバレッジ: 2.6434
3つの積: 0.0964
ROE（期末の自己資本）: 0.0964
ROE（平均の自己資本）: 0.1015
ROE（決算サマリーの列）: 0.101


期末の自己資本で割ったROEは約9.64%で、デュポン分解の積と一致します。決算サマリーの `ROE` 列の0.101は、前期末と当期末の自己資本の平均で割った約10.15%を丸めた値です。

やってみる

前期末の自己資本（期首）で割ったROEを計算します。約10.71%です。

In [23]:
# ここに書きます

## 2-6　PER・PBR・配当利回り

実行する

In [24]:
per = price / cur["EPS"] if pd.notna(cur["EPS"]) and cur["EPS"] > 0 else float("nan")
pbr = price * shares / cur["ShEq"] if cur["ShEq"] > 0 and shares > 0 else float("nan")
dividend_yield = cur["DivAnn"] / price
roe_end = net_income / equity

print("PER:", round(per, 2))
print("PBR:", round(pbr, 4))
print("配当利回り:", round(dividend_yield, 4))
print("PER × ROE（期末）:", round(per * roe_end, 4))

PER: 9.4
PBR: 0.9065
配当利回り: 0.0342
PER × ROE（期末）: 0.9065


PERは約9.40倍、PBRは約0.9065倍、配当利回りは約3.42%です。PERの `EPS` は2026年3月期の実績で、予想ではありません。PER×ROEは約0.9065倍で、PBRとほぼ一致します。わずかな差は、EPSが期中平均の株数で、BPSが期末の株数で計算されているためです。

やってみる

評価日を2026年6月1日（終値2,905.5円）に変えて、PERとPBRを計算します。PERは約9.84倍、PBRは約0.9486倍です。

In [25]:
# ここに書きます

# 3. 最初から実行し直す

1. notebook上部の Restart を押します。kernelの変数がすべて消えます。
2. Run All を押します。上のcellから順に、全部が実行されます。

2節ではもう一度APIから取得するので、API keyの入力欄が出ます。1-6で `price = 1500` を足したままなら、1-6以降の結果が変わります。

# 4. 発展：日産自動車とイオン

2節のcodeの銘柄コードを、日産自動車（72010）またはイオン（82670）に変えて計算します。日産自動車は2026年3月期、イオンは2026年2月期が最新の通期決算です。

- 日産自動車は純利益が負で、EPSは−152.58円です。PERは約−2.15倍と計算できますが、倍率として比べる意味はありません。
- イオンは2025年に1株を3株に分割しました。2025年2月期のBPS 1,218.63円は分割前、2026年2月期の440.40円は分割後の株数で計算されています。

In [26]:
# ここに書きます